specification/cognitiveservices/OpenAI.Inference/models/chat.completions.tsp

```diff
@@ -0,0 +1,197 @@
+import "@typespec/rest";
+import "@typespec/http";
+import "./completions.common.tsp";
+
+using TypeSpec.Rest;
+using TypeSpec.Http;
+
+@doc("A description of the intended purpose of a message within a chat completions interaction.")
+enum ChatRole {
+    @doc("The role that instructs or sets the behavior of the assistant.")
+    @projectedName("json", "system")
+    system,
+
+    @doc("The role that provides responses to system-instructed, user-prompted input.")
+    @projectedName("json", "assistant")
+    assistant,
+
+    @doc("The role that provides input for chat completions.")
+    @projectedName("json", "user")
+    user,
+}
+
+@doc("A single, role-attributed message within a chat completion interaction.")
+model ChatMessage {
+    @doc("The role associated with this message payload.")
+    @projectedName("json", "role")
+    role: ChatRole;
+
+    @doc("The text associated with this message payload.")
+    @projectedName("json", "content")
+    content?: string;
+}
+
+@doc("""
+The configuration information for a chat completions request.
+Completions support a wide variety of tasks and generate text that continues from or "completes"
+provided prompt data.
+""")
+model ChatCompletionsOptions {
+    @doc("""
+    The collection of context messages associated with this chat completions request.
+    Typical usage begins with a chat message for the System role that provides instructions for
+    the behavior of the assistant, followed by alternating messages between the User and
+    Assistant roles.
+    """)
+    @projectedName("json", "messages")
+    messages?: ChatMessage[];
```

Comment:
Nice find! I tried it and verified that you need to provide `messages` to either OAI or AOAI.

I also double-checked `prompt` for `/completions` and found some inconsistencies between OAI and AOAI:
1. OAI does not require `prompt` and will follow the described, implicit `<|endoftext|>` token; AOAI *does* require prompt and will fail if you post without it
2. OAI allows either a single string or array of strings; AOAI only allows an array of strings

I'll follow up on those separately.

Reply:
One interesting consequence of this that bears discussion: making it a required parameter promotes it to a required constructor parameter with default code generation. I'm not sure where we stand on balancing the considerations there; on one hand, it makes sense that required data inputs are required on the constructor to avoid accidental omission; on the other, it's a bit awkward to promote and enforce it this way when it'll still be checked as part of the request.

I'll make it conform for now (restore a custom constructor that doesn't require `IEnumerable<ChatMessage>` in the .NET PR) but this is probably a topic for API review.